pkg/compactor/block_upload.go

```diff
@@ -262,22 +272,58 @@ func (c *MultitenantCompactor) UploadBlockFile(w http.ResponseWriter, r *http.Re
 	w.WriteHeader(http.StatusOK)
 }
 
-func (c *MultitenantCompactor) completeBlockUpload(ctx context.Context, logger log.Logger, userBkt objstore.Bucket, blockID ulid.ULID, meta metadata.Meta) error {
+func (c *MultitenantCompactor) completeBlockUpload(ctx context.Context, logger log.Logger, userBkt objstore.Bucket, blockID ulid.ULID, meta metadata.Meta) {
 	level.Debug(logger).Log("msg", "completing block upload", "files", len(meta.Thanos.Files))
 
+	// start go routine that updates validation file timestamp once per minute
+	ch := make(chan string)
+	var wg sync.WaitGroup
+	wg.Add(1)
+	ctx, cancel := context.WithCancel(ctx)
+	go c.periodicValidationUpdater(ctx, logger, blockID, userBkt, ch, &wg, cancel)
```

Comment:
I think it's pointless to create the cancellation function in the caller of `c.periodicValidationUpdater`, since it's an implementation detail of the latter. I.e., it's better created in `c.periodicValidationUpdater`.
```suggestion
	go c.periodicValidationUpdater(ctx, logger, blockID, userBkt, ch, &wg)
```